blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx

```diff
@@ -0,0 +1,312 @@
+
+As we saw in **[How to Store Vibration Sensor Data | Part 1](/blog/how-to-store-vibration-sensor-data)**, it's beneficial to divide the data into chunks for more efficient storage and querying.
+In this example, we generate 1 second of data at a time (1,000 samples at 1 kHz), that we'll store as a single entry in ReductStore.
+
+### Calculating Metrics
+
+```python
+def calculate_metrics(signal: np.ndarray) -> tuple:
+    rms = np.sqrt(np.mean(signal**2))
+    peak_to_peak = np.max(signal) - np.min(signal)
+    crest_factor = np.max(np.abs(signal)) / rms
+    return rms, peak_to_peak, crest_factor
+```
+
+We calculate three common metrics for our signal: RMS (Root Mean Square), Peak-to-Peak, and Crest Factor.
+
+### Packing Binary Data
+
+```python
+def pack_data(signal: np.ndarray) -> bytes:
+    fmt = f">{len(signal)}f"
+    return struct.pack(fmt, *signal)
+```
+
+This function uses the `struct` module to pack our numpy array into a binary format, specifically with the format string `">1000f"` (more details on this below).
+You may be wondering why we don't use numpy's `tobytes()` method. While `tobytes()` is convenient, it offers limited control over the byte format, 
+which can lead to compatibility problems when reading the data on different devices.
+
+The `struct` module, on the other hand, allows us to specify byte order and data type, preserving consistent data representation and avoiding compatibility problems.
+The format string `">1000f"` is explained as follows, based on the **[struct module documentation](https://docs.python.org/3/library/struct.html#struct-format-strings)**:
+
+- `>` indicates big-endian byte order, with the most significant byte (leftmost byte) stored first, which is common in network communications.
+- `1000` is the number of elements in the array (1000 samples).
+- `f` is the data type (float) for each element, with a default size of 4 bytes.
+
+The choice of binary data depends on your specific requirements, if you are using specific hardware or software that requires a different format, you can adjust the format string accordingly.
+
+Some restricted embedded systems may require a specific byte order or data type, so it's important to understand the format requirements of your architecture.
+
+### Storing Data in ReductStore
+
+```python
+HIGH_RMS = 1.0
+HIGH_CREST_FACTOR = 3.0
+HIGH_PEAK_TO_PEAK = 5.0
+
+async def store_data(
+    bucket: Bucket,
+    timestamp: int,
+    packed_data: bytes,
+    rms: float,
+    peak_to_peak: float,
+    crest_factor: float,
+):
+    labels = {
+        "rms": "high" if rms > HIGH_RMS else "low",
+        "peak_to_peak": "high" if peak_to_peak > HIGH_PEAK_TO_PEAK else "low",
+        "crest_factor": "high" if crest_factor > HIGH_CREST_FACTOR else "low",
+    }
+    await bucket.write("sensor_readings", packed_data, timestamp, labels=labels)
+```
+
+This is where we store our packed data, along with labels that indicate whether each metric is high or low. 
```

Comment:
data is too abstract and someone could not get that we actually write  a chunk. Lets say packed `chunk of data`.